### ISSUE TITLE
Cannot make multiple pulses work

### ISSUE THREAD

**Issue by joefowler** [ggggggggg/ModelTES.jl#9]
Galen, I just merged my fork with yours, and I still get an error when I try either of the 2nd or 3rd biasesTES in the following code:

```julia
using ModelTES, PyPlot

bt1 = ModelTES.pholmes(48e-9, 0.20)
bt2 = ModelTES.lowEpix()
bt3 = ModelTES.LCLSII(52e-9)

sampletime = 1e-7
pulsetimes = [1e-4, 2e-4]

clf()
for bt in (bt1, bt2, bt3)
    outmany = ModelTES.pulses(6000, sampletime, bt, [1000,1000], pulsetimes);
    plot(outmany.I[1]-outmany.I)
end
```

Specifically, a NaN-based error:

```
ERROR: LoadError: DomainError:
 in nan_dom_err at ./math.jl:196 [inlined]
 in ^(::Float64, ::Float64) at ./math.jl:355
 in dT at /Users/fowlerj/.julia/v0.5/ModelTES/src/stochastic_integration.jl:7 [inlined]
 in (::ModelTES.BiasedTES{ModelTES.ShankRIT})(::Float64, ::Array{Float64,1}, ::Array{Float64,1}) at /Users/fowlerj/.julia/v0.5/ModelTES/src/stochastic_integration.jl:88
 in perform_step! at /Users/fowlerj/.julia/v0.5/OrdinaryDiffEq/src/integrators/low_order_rk_integrators.jl:224 [inlined]
 in perform_step! at /Users/fowlerj/.julia/v0.5/OrdinaryDiffEq/src/integrators/low_order_rk_integrators.jl:204 [inlined]
 in solve!(::OrdinaryDiffEq.ODEIntegrator{OrdinaryDiffEq.Tsit5,Array{Float64,1},Float64,Float64,Float64,Array{Array{Float64,1},1},DiffEqBase.ODESolution{Array{Array{Float64,1},1},Array{Float64,1},Array{Array{Array{Float64,1},1},1},DiffEqBase.ODEProblem{Array{Float64,1},Float64,true,ModelTES.BiasedTES{ModelTES.ShankRIT}},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{ModelTES.BiasedTES{ModelTES.ShankRIT},Array{Array{Float64,1},1},Array{Float64,1},Array{Array{Array{Float64,1},1},1},OrdinaryDiffEq.Tsit5Cache{Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Float64,1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}},Array{Float64,1},ModelTES.BiasedTES{ModelTES.ShankRIT},Void,OrdinaryDiffEq.Tsit5Cache{Array{Float64,1},Array{Float64,1},Array{Float64,1},Array{Float64,1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}},OrdinaryDiffEq.DEOptions{Float64,Float64,Float64,Float64,OrdinaryDiffEq.#ODE_DEFAULT_NORM,DiffEqBase.CallbackSet{Tuple{},Tuple{DiffEqBase.DiscreteCallback{ModelTES.##11#15{Array{Float64,1}},ModelTES.##12#16{ModelTES.BiasedTES{ModelTES.ShankRIT},Dict{Float64,Int64}}}}},OrdinaryDiffEq.#ODE_DEFAULT_ISOUTOFDOMAIN,OrdinaryDiffEq.#ODE_DEFAULT_PROG_MESSAGE,OrdinaryDiffEq.#ODE_DEFAULT_UNSTABLE_CHECK,DataStructures.BinaryHeap{Float64,DataStructures.LessThan},Void}}) at /Users/fowlerj/.julia/v0.5/OrdinaryDiffEq/src/solve.jl:250
 in #solve#45(::Array{Any,1}, ::Function, ::DiffEqBase.ODEProblem{Array{Float64,1},Float64,true,ModelTES.BiasedTES{ModelTES.ShankRIT}}, ::OrdinaryDiffEq.Tsit5, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at /Users/fowlerj/.julia/v0.5/OrdinaryDiffEq/src/solve.jl:7
 in (::DiffEqBase.#kw##solve)(::Array{Any,1}, ::DiffEqBase.#solve, ::DiffEqBase.ODEProblem{Array{Float64,1},Float64,true,ModelTES.BiasedTES{ModelTES.ShankRIT}}, ::OrdinaryDiffEq.Tsit5, ::Array{Any,1}, ::Array{Any,1}, ::Array{Any,1}, ::Type{Val{true}}) at ./<missing>:0 (repeats 2 times)
 in #pulses#9(::Float64, ::OrdinaryDiffEq.Tsit5, ::Float64, ::Float64, ::Function, ::Int64, ::Float64, ::ModelTES.BiasedTES{ModelTES.ShankRIT}, ::Array{Int64,1}, ::Array{Float64,1}) at /Users/fowlerj/.julia/v0.5/ModelTES/src/ModelTES.jl:352
 in pulses(::Int64, ::Float64, ::ModelTES.BiasedTES{ModelTES.ShankRIT}, ::Array{Int64,1}, ::Array{Float64,1}) at /Users/fowlerj/.julia/v0.5/ModelTES/src/ModelTES.jl:342
 in macro expansion; at /Users/fowlerj/Microcal/Nonlinearity/Demos/swetz_bes_demo.jl:12 [inlined]
 in anonymous at ./<missing>:?
 in include_from_node1(::String) at ./loading.jl:488
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
while loading /Users/fowlerj/Microcal/Nonlinearity/Demos/swetz_bes_demo.jl, in expression starting on line 11
```

Unfortunately, now that you installed a package with a totally inscrutable interface, I'm helpless here. And I promised Dan some kind of demo plot quickly. Ideas??

**Comment by joefowler** [comment]
I can say that the NaN error arises only for certain models and certain energies. So for example, the following code makes the plot that I want, then it _succeeds_ on the next-to-last line but _fails_ on the last, even though they are the same `BiasedTES` object and differ only in the energy.

```julia
using ModelTES, PyPlot

bt1 = ModelTES.pholmes(12e-9, 0.20)
bt2 = ModelTES.lowEpix()
bt3 = ModelTES.highEpix()
bt4 = ModelTES.LCLSII(52e-9)

sampletime = 1e-6
pulsetimes = [3e-4, 8e-4]

N = 4000
t = 1e6*(collect(0:N-1)*sampletime-pulsetimes[1])
fast = ModelTES.pulses(N, sampletime, bt1, [525,525], pulsetimes)
slow = ModelTES.pulses(N, sampletime, bt3, [2000,2000], pulsetimes)

clf()
plot(t, 8+1e6*(fast.I[1]-fast.I), color="r")
plot(t, 1e6*(slow.I[1]-slow.I), color="b")
ylabel("Δ Current (μA)")
xlabel("Time (μs)")

fine = ModelTES.pulses(N, sampletime, bt4, [20.0,20.0], pulsetimes)
fails = ModelTES.pulses(N, sampletime, bt4, [200.0,200.0], pulsetimes)
```

**Comment by joefowler** [comment]
Sorry, I forgot to clarify: I have what I need to make Dan's plot before the weekend.

**Comment by ggggggggg** [comment]
I have a fix for this. I think whats happening is that the adaptive solver is increasing the timesteps as it goes, then a new pulse arrives, and it isn't reducing it's timestep. This cause it to explore such a large timestep that the temperature goes negative and you cant take `-1.^1.5` in julia without a domain error.

If I just use `T=max(0.0, T)` it seems to work. I posted in the DifferentialEquations.jl gitter asking for advice on a better solution.

**Comment by ggggggggg** [comment]
Again, I have run into a feature that is not fully ready in `DifferentialEquations.jl`. I'm supposed to use `DifferentialEquations.modify_proposed_dt!(integrator,DifferentialEquations.get_proposed_dt(integrator)/dtsolver)` to control the timestep, but those functions are not implemented yet. There are just placeholder functions saying they aren't implemented yet. And the docs get the function names slightly wrong.

**Comment by ggggggggg** [comment]
Fixed by 7c9f2aa
